src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -115,10 +115,17 @@ private function createType(DocType $type, bool $nullable, ?string $docType = nu
 
             [$phpType, $class] = $this->getPhpTypeAndClass((string) $fqsen);
 
+            $collection = \in_array($class, [\Traversable::class, \Iterator::class, \IteratorAggregate::class, \ArrayAccess::class, \Generator::class], true);
```

Comment:
shouldn't we at least support the Collection and ReadableCollection interfaces of doctrine/collections there, to limit the impact of the BC break ?

Reply:
Similar is being done in https://github.com/symfony/symfony/pull/54661